File: 2021-rabbitmq-jpa/domain/jpa/src/main/java/sample/domain/jpa/repository/CommentRepository.java
package sample.domain.jpa.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import sample.domain.jpa.model.ArticleEntity;
import sample.domain.jpa.model.CommentEntity;

import java.util.Optional;

@Repository
public interface CommentRepository extends JpaRepository<CommentEntity, Long> {
    Optional<CommentEntity> findFirstByArticleOrderByIdDesc(ArticleEntity article);
}

File: 2021-rabbitmq-jpa/domain/jpa/src/main/kotlin/sample/domain/jpa/repository/CommentRepository.kt
package sample.domain.jpa.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import sample.domain.jpa.model.article.ArticleEntity
import sample.domain.jpa.model.comment.CommentEntity
import java.util.*

@Repository
interface CommentRepository : JpaRepository<CommentEntity, Long> {
    fun findFirstByArticleOrderByIdDesc(article: ArticleEntity): Optional<CommentEntity>
}
